lib/utils/reify-cmd.js

```diff
@@ -0,0 +1,24 @@
+// This is the base for all commands whose execWorkspaces just gets
+// a list of workspace names and passes it on to new Arborist() to
+// be able to run a filtered Arborist.reify() at some point.
+
+const BaseCommand = require('../base-command.js')
+const getWorkspaces = require('../workspaces/get-workspaces.js')
+class ReifyCmd extends BaseCommand {
+  static get params () {
+    return [
+      'workspaces',
+      'workspace',
+    ]
+  }
+
+  execWorkspaces (args, filters, cb) {
+    getWorkspaces(filters, { path: this.npm.localPrefix })
+      .then(workspaces => {
+        this.workspaces = [...workspaces.keys()]
+        this.exec(args, cb)
+      })
```

Comment:
`cb` will never fire if `getWorkspaces` rejects e.g. `No workspaces found${msg}` https://github.com/npm/cli/blob/6e4c1f38bb5d3535c7e565c7991885809fe36700/lib/workspaces/get-workspaces.js#L30

```suggestion
      }, cb)
```

Reply:
The other commands use `.catch` instead of the second parameter error handler method. We don't really use the second parameter method for errors here (both work but consistency is also nice)

https://github.com/npm/cli/blob/latest/lib/repo.js#L36

Reply:
@wraithgar Sure, but that will cause `cb` to fire a second time for a single invocation of `execWorkspaces ` if `cb` happens to reject/throw. Control should never return to this function after `cb` is called once. 

e.g.

```js
function task(cb) {
   Promise.resolve().then(() => cb(null, true)).catch(cb)
}

task((err, value) => {
  console.log('task callback', err, value)
  throw new Error('oops')
})

// task callback null true
// task callback Error: oops
```

Using second argument to `.then` will turn that into an unhandled rejection, but that's probably the correct behaviour instead of calling the callback a second time when there's an error in the callback that was called and had an error. i.e. callback should not be invoked again to handle/report errors in itself.

Reply:
The real solution is to move off of callbacks, which is on our refactoring TODO list. For now I think we should be consistent.  The callback it is being passed is our error-handler, which already has a "callback called twice" checker.

Reply:
> The real solution is to move off of callbacks

👍 

>  The callback it is being passed is our error-handler, which already has a "callback called twice" checker.

Sure yep, though note this is only set up for commands executed via cli, there'd be no such safety for internal calls to these methods, I don't see any so yeah perhaps a non-issue.